junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/MutableTestExecutionSummary.java

```diff
@@ -192,13 +194,13 @@ public void printFailuresTo(PrintWriter writer) {
 
 	@Override
 	public List<Failure> getFailures() {
-		return new ArrayList<>(failures);
+		return Collections.unmodifiableList(new ArrayList<>(failures));
```

Comment:
Imho, yes. "JUnit 5" is the only "client" affected by the change, anyway. `MutableTestExecutionSummary` is package private and therefore `@API(Internal)`.